pkg/experimental/pkg_filegroup.bzl

```diff
@@ -578,6 +591,83 @@ pkg_mkdirs = rule(
             The default is `dir`.
             """,
             default = "dir",
+            # TODO(nacl): use "values" here
+        ),
+    },
+)
+
+def _pkg_mklinks_impl(ctx):
+    _validate_attr(ctx.attr.attrs)
+    return [
+        PackageSymlinkInfo(
+            link_map = ctx.attr.links,
+            attrs = ctx.attr.attrs,
+            section = ctx.attr.section,
+        ),
+    ]
+
+pkg_mklinks = rule(
+    doc = """pkg_filegroup-like-rule for symlink management.
+
+    This rule results in the creation of one or more symbolic links a package.
+    The link may point to a location outside of it.
+
+    """,
+    implementation = _pkg_mklinks_impl,
+    attrs = {
+        "links": attr.string_dict(
+            doc = """Link mappings to create.
+
+            The keys of this dict are the link names, the values are the source
+            files.
+
+            Note that relative paths are important here: the package builder MAY
+            adjust the "link name" of the pair (i.e. make paths absolute), but
+            it MUST NOT attempt to do anything with the "source" part.
+
+            """,
+            mandatory = True,
+        ),
+        "attrs": attr.string_list_dict(
+            doc = """Attributes to set for the output targets.
+
+            Must be a dict of:
+
+            ```
+            "unix" : [
+                "Four-digit octal permissions string (e.g. "0755") or "-" (don't change from what's provided),
+                "User Id, or "-" (use current user)",
+                "Group Id, or "-" (use current group)",
+            ]
+            ```
+
+            The permissions value defaults to "0777".  The user/group values
+            default to "-".
+            """,
+            default = {"unix": ["0777", "-", "-"]},
+        ),
+        "section": attr.string(
+            doc = """Symlink section mapping.
+
+            Legal values for this attribute are:
```

Comment:
We should consider a way to not repeat this each time. But that can be another pass.

You don't really enforce this, do you? It's only the rpm builder which will.  Perhaps a more interesting way is to remove any restrictions on section, and change it to be a universal tag which is used by rpm, deb, whatever as they find appropriate. We could have each package builder filter the rules it does not understand. OTOH, that could lead to unintended missing specs, so probably not.

Reply:
Finally got some time to look at all this.  Sorry for the delays.

> We should consider a way to not repeat this each time. But that can be another pass.

Yeah, some sort of common attributes list wouldn't be a bad idea.

> You don't really enforce this, do you? It's only the rpm builder which will. Perhaps a more interesting way is to remove any restrictions on section, and change it to be a universal tag which is used by rpm, deb, whatever as they find appropriate. We could have each package builder filter the rules it does not understand. OTOH, that could lead to unintended missing specs, so probably not.

Yeah, I really don't do much other than check for whether or not values are mentioned in a list.

Your proposal is indeed more flexible, along with your concerns.  I get the feeling they could perhaps be mitigated if there was a way to debug them easily -- perhaps a custom [command-line](https://docs.bazel.build/versions/master/skylark/config.html) flag and/or a rule attribute that increases logging or causes failures?

If you think the above is a good idea, I'll create some tasks to work on this.

Reply:
Hi @nacl - did you ever get a chance to create a follow-up tasks/issue to decide on how to reduce this repetition? I can do so if not.

Reply:
Yup, check out #191.